unit_test/entities/controller/test_controller_class.py

```diff
@@ -82,6 +84,7 @@ def test_controller_init(self):
             temperature_return=20.0,
             temperature_supply=30.0,
             power=1.0,
+            marginal_costs=0.1,
```

Comment:
Missing arange statement above.

Can you check all the tests of this class? The don't follow the agreed upon pattern!

Reply:
Why don't you use the properties defined in the setUp? You could even create the ControllerProducer, etc in the SetUp.

Reply:
fixed